restore.py

```diff
@@ -203,29 +203,49 @@ def restore_partitions():
                 (tool, restore_partition) = restore_partitions()
                 dest_fs = util.TempMount(restore_partition, 'restore-dest-')
 
+            if diskutil.is_raid(disk_device):
+                backend.waitForSWRAIDSync(disk_device)
+
+                # umount and re-mount partitions to flush the cache
+                # umount all partitions first before re-mounting
+                if efi_mounted and "esp" in mounts:
```

Comment:
Rather than adding more `efi_mounted` conditionals, I think it would be better to add a commit before this that cleans up this function and drops all the conditional EFI stuff from this function. XS9 only supports upgrade from UEFI systems, therefore restores will always be back to a UEFI system.